### PR TITLE
add StringField, fix todos on MarkdownField

### PR DESCRIPTION
**Pull request by davisagli**
Revisions to https://github.com/SFDO-Tooling/sfdo-template-helpers/pull/23 from @cdcarter

### CODE REVIEW

File: sfdo_template_helpers/fields/markdown.py [hunk 1]
```diff
@@ -70,19 +66,36 @@ class MarkdownFieldMixin:
...
         # Only include kwarg if it's not the default
-        if self.property_suffix != DEFAULT_SUFFIX:
+        if self.property_suffix != MarkdownField.property_suffix:
             kwargs["property_suffix"] = self.property_suffix
+        if not self.is_safe:
+            kwargs["is_safe"] = self.is_safe
+        if self.allowed_tags != MarkdownField.allowed_tags:
+            kwargs["allowed_tags"] = self.allowed_tags
+        if self.allowed_attrs != MarkdownField.allowed_attrs:
+            kwargs["allowed_attrs"] = self.allowed_attrs
```
Comment (wlonk) [inline sfdo_template_helpers/fields/markdown.py]: I have a small concern about these three conditionals: they have a long-distance interaction with the defaults defined above on lines 83-85, such that if either change, the other should as well. Can we, should we, do we want to express that in code? Like `if not self.is_safe == IS_SAFE_DEFAULT` and use the same `IS_SAFE_DEFAULT` on line 83?

File: tox.ini [hunk 2]
```diff
@@ -21,3 +21,8 @@ omit = sfdo_template_helpers/urls.py
 
 [coverage:report]
 fail_under = 100
+
+[pytest]
+DJANGO_SETTINGS_MODULE = tests.settings
+python_files = test_*.py
```
Comment (wlonk) [inline tox.ini]: This is the default, right?

File: sfdo_template_helpers/fields/markdown.py (hunk 1 above)
Comment (davisagli) [reply sfdo_template_helpers/fields/markdown.py]: I see your point, but I think these are unlikely to change and we'll scrutinize them if they do.

Comment (cdcarter) [reply sfdo_template_helpers/fields/markdown.py]: if you have the custom Fields manual up while editing this code, you'll have the understanding that if you touch a constructor for a Field, you need to touch its deconstructor as well. 

But you're totally right, I do wish there was a better way to express the defaults for a django field when also using mutable collections. the default should probably be a frozenset too... i'll thnk about it...

File: sfdo_template_helpers/fields/markdown.py [hunk 3]
```diff
@@ -28,14 +24,14 @@ def __get__(self, instance, owner):
         raw_md = instance.__dict__[self.field.name]
         if raw_md is None:
             return ""
-        safer = mark_safe if self.field.is_safe else supplier
```
Comment (cdcarter) [inline sfdo_template_helpers/fields/markdown.py]: 😅 this seemed like a good idea at the time, so thank you for making it actually understandable :)

File: sfdo_template_helpers/fields/string.py [hunk 4]
```diff
@@ -11,6 +12,8 @@ class StringField(models.TextField):
...
     That's all."""
 
-    def formfield(self, **kwargs):
-        # skip TextField's override of models.Field.formfield
-        return models.Field.formfield(self, **kwargs)
+
+# Make sure StringField has its own mapping in FORMFIELD_FOR_DBFIELD_DEFAULTS
+# so that it takes precedence over its base class TextField.
+from django.contrib.admin.options import FORMFIELD_FOR_DBFIELD_DEFAULTS
+FORMFIELD_FOR_DBFIELD_DEFAULTS[StringField] = {'widget': widgets.AdminTextInputWidget}
```
Comment (cdcarter) [inline sfdo_template_helpers/fields/string.py]: 🎉

**Pull request merged**